File: Barking Dog/BarkingDog.rb
def BarkingDog(barking, hourOfTheDay)
    if (barking == true)
            if ((hourOfTheDay==23 || hourOfTheDay<8) && hourOfTheDay>=0 )
                return true;
            end
        return false;
    end
end
BarkingDog(true,77)

File: Barking Dog/BarkingDog.php
<?php
function bark($barking,$houroftheday){
    if ($barking == true){
        if (($houroftheday==23 || $houroftheday<8) && $houroftheday>=0)
            echo "true";
        else{
            echo "false";
        }
    }else{
        echo "false";
    }
}
bark(true,0);
